### ISSUE TITLE
Shift drag ZoomTo box

### ISSUE THREAD

**Issue by FredFunk** [openseadragon/openseadragon#1240]
Does the functionality exist to drag a box over an area of the image and zoom to that area? Something similar to holding Shift key down and dragging mouse (shift+drag) to zoom to bounding box?

Animated example here:
https://github.com/mapbox/mapbox-gl-js/issues/1038

https://github.com/Leaflet/Leaflet/blob/master/src/map/handler/Map.BoxZoom.js

**Comment by iangilman** [comment]
It does not, but it shouldn't be too hard to build and would make a great plugin! This plugin might be a good starting point: 

https://github.com/picturae/openseadragonselection

**Comment by sanioka** [comment]
Popular IIIF viewer like OpenLayers (7000+ github stars) also has this feature out of the box.
https://openlayers.org/en/latest/apidoc/module-ol_interaction_DragZoom-DragZoom.html

demo
https://openlayers.org/en/latest/examples/iiif.html

Do you have any plan to implement it to OpenSeadragon?

**Comment by iangilman** [comment]
As far as I know, there has been no progress on this. I still think in the context of OpenSeadragon it makes more sense as a plugin, but I certainly would love to see it happen! Any interest in building it, @sanioka?

**Comment by msalsbery** [comment]
I agree...it can be done relatively easily as a plugin. The [openseadragon-viewerinputhook plugin](https://github.com/openseadragon-imaging/openseadragon-viewerinputhook) (or equivalent) was meant for stuff like this, making it easy to mix in new functionality with the existing viewer functionality.
